src/Microsoft.Performance.Toolkit.Engine.Tests/ToolkitEngineCreateInfoTests.cs

```diff
@@ -18,5 +24,75 @@ public void DefaultRuntime_Set()
             string expectedName = typeof(EngineCreateInfo).Assembly.GetName().Name;
             Assert.AreEqual(expectedName, sut.RuntimeName);
         }
+
+        [TestMethod]
+        [IntegrationTest]
+        public void DefaultProcessorOptionsResolver_Set()
+        {
+            var sut = new EngineCreateInfo(DataSourceSet.Create().AsReadOnly());
+            Assert.IsNotNull(sut.OptionsResolver, "Options Resolver is null when a default is expected");
+            Assert.AreEqual(typeof(GlobalProcessingOptionsResolver), sut.OptionsResolver.GetType(), "Options Resolver is not of type GlobalProcessingOptionsResolver");
+        }
+
+        [TestMethod]
+        [IntegrationTest]
+        public void GlobalProcessorOptionsResolver_Test()
+        {
+            var processorOptions = new ProcessorOptions(
+                new[]
+                {
+                    new OptionInstance(
+                        new Option('s', "test1"),
+                        "arg1"),
+                });
+
+            var dsg = new DataSourceGroup(new[] { new FileDataSource("sample") }, new DefaultProcessingMode());
+            ProcessingSourceReference.TryCreateReference(typeof(Source123DataSource), out var ps);
+
+            var sut = new EngineCreateInfo(DataSourceSet.Create().AsReadOnly());
+            sut.WithProcessorOptions(processorOptions);
+
+            Assert.IsNotNull(sut.OptionsResolver, "Options Resolver is null when a default is expected");
+            Assert.AreEqual(typeof(GlobalProcessingOptionsResolver), sut.OptionsResolver.GetType(), "Options Resolver is not of type GlobalProcessingOptionsResolver");
+            
+            Assert.AreEqual(processorOptions, sut.OptionsResolver.GetProcessorOptions(ps.Guid, dsg), "Expected processor options were not returned" );
+        }
+
+        [TestMethod]
+        [IntegrationTest]
+        public void ProcessingSourceOptionsResolver_Test()
+        {
+            var processorOptions = new ProcessorOptions(
+                new[]
+                {
+                    new OptionInstance(
+                        new Option('s', "test1"),
+                        "arg1"),
+                });
+
+            ProcessingSourceReference.TryCreateReference(typeof(Source123DataSource), out var psr);
```

Comment:
I believe we can avoid creating the PSRs in this file just like the other test file, right?